Algorithms/Encoders/NYSIISEncoder.cs

```diff
@@ -0,0 +1,197 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for NYSIIS encoding strings.
+    /// </summary>
+    public class NYSIISEncoder
+    {
+        private static char[] Vowels = new char[] { 'A', 'E', 'I', 'O', 'U' };
+
+        /// <summary>
+        /// Creates a new instance of the NYSIISEncoder.
+        /// </summary>
+        public NYSIISEncoder() {}
+
+        /// <summary>
+        /// Encodes a string using the NYSIIS Algorithm
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The NYSIIS encoded string (all uppercase)</returns>
+        public string Encode(string text)
```

Comment:
Codacy found an issue: [Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)